### DISCUSSION TITLE
OperationId (endpoint) has Changed for the Same Product?

### DISCUSSION THREAD

**Discussion by victorthemtt** [ercot/api-specs#57] [Q&A] [answered]
Hello, 

I don't know if anybody posted this but I found that the endpoint for some data has changed from the documentation I downloaded in September to the one in October. See the screenshots that show NP3-233-CD as an example. First image is the old version API documentation and the second is the new one. Why is this happening? 

![image](https://github.com/user-attachments/assets/9fe4f329-ef1e-4ce9-8334-5b11240bda5b)
![image](https://github.com/user-attachments/assets/b0e8447e-9592-4d56-8217-978ab451945c)

Thank you for your response. 
Victor Mao

**Comment by mackermann-ercot** [comment, marked as answer]
Hi Victor,

You are correct in the fact that the "operationId" values may change over time.  It is an internal identifier used by our API management system, but it not intended for any particular use outside of internal Public API processes.  The OpenAPI specification requires that information to be included in the document.  The identifier itself is generated from the system in order of report generation, thus, when we add reports, those numbers may get shifted (since the system generates in alphabetical order).  

Note that in future releases, we will be changing the operationId to reflect the actual endpoint name in lieu of a simple sequential order.

Regards,
mackermann-ercot

**Reply by victorthemtt** [reply]
Hi Mackermann, 

Thank you for your kind response. So, you mean this should not impact the API request URL and the data retrieval process, right? 

I got another issue while retrieving the data. I encountered the 401 Unauthorized error after successfully retrieving data in the pagination process, as the output shows: 
"HTTP response body: b'{ "statusCode": 401, "message": "Unauthorized. Access token is missing or invalid." }'"
My login credentials should be correct since I can get the data before this error, so I'm wondering if this is due to: 
1. The rate limiting of the API request. Are there any rate limiting?
2. Access token expired since the pagination took very long (1 hour). I estimate an hour is roughly the valid duration for the access token. 

Appreciate your help, 
Victor Mao

**Comment by CaffeineLab** [comment]
You may not get the official answer until Monday, but you should start by limiting your requests to 1 every 2 seconds.  Yes.  It is slow. 

1 - Rate limiting is discussed here:  https://github.com/ercot/api-specs/discussions/44 but essentially is:

>  It is indeed 30 calls per 60 seconds which would be 1 request every 2 seconds,

2 - I haven't personally run into access tokens expiring even when pulling all reports for multiple hours in one requests session.
